ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala

```diff
@@ -211,8 +211,14 @@ object KeyValueCommitting {
       transactionEntry: DamlTransactionEntry
   ): Set[DamlStateKey] = {
     val outputs = Set.newBuilder[DamlStateKey]
-    transactionEntry.getTransaction.getNodesList.asScala.foreach { case node =>
-      val nodeVersion = TransactionCoder.decodeVersion(node.getVersion) match {
+    val txVersion =
+      TransactionCoder.decodeVersion(transactionEntry.getTransaction.getVersion) match {
+        case Right(value) => value
+        case Left(err) => throw Err.DecodeError("Transaction", err.errorMessage)
+      }
+
+    transactionEntry.getTransaction.getNodesList.asScala.foreach { node =>
+      val nodeVersion = TransactionCoder.decodeNodeVersion(txVersion, node) match {
         case Right(value) => value
         case Left(err) => throw Err.DecodeError("Node", err.errorMessage)
       }
```

Comment:
We agreed we wouldn't write a test case here yet.

Reply:
:disappointed: